### PR TITLE
Cm boundary groups

### PR DESCRIPTION
**Pull request by jeffotterpohl**

### CODE REVIEW

File: source/DSCResources/DSC_CMBoundaryGroups/DSC_CMBoundaryGroups.psm1 [hunk 1]
```diff
@@ -0,0 +1,364 @@
...
+                }
+
+                foreach ($member in $includes)
+                {
+                    if (Get-CMBoundary -BoundaryName $member)
+                    {
+                        if ($state.Boundaries -notcontains $member)
+                        {
+                            Write-Verbose -Message ($script:localizedData.AddingBoundary -f $BoundaryGroup, $member)
+                            Add-CMBoundaryToGroup -BoundaryName $member -BoundaryGroupName $BoundaryGroup
```
Comment (rchristman89) [inline source/DSCResources/DSC_CMBoundaryGroups/DSC_CMBoundaryGroups.psm1:146]: Check if there are issues with same name different type.

File: source/DSCResources/DSC_CMBoundaryGroups/DSC_CMBoundaryGroups.psm1 [hunk 2]
```diff
@@ -0,0 +1,364 @@
...
+                    }
+                    else
+                    {
+                        $errorMsg += ($script:localizedData.BoundaryAbsent -f $member)
+                        Write-Verbose -Message ($script:localizedData.BoundaryAbsent -f $member)
+                    }
+                }
+            }
+
+            if (($Boundaries) -or ($BoundariesToExclude))
```
Comment (rchristman89) [inline source/DSCResources/DSC_CMBoundaryGroups/DSC_CMBoundaryGroups.psm1:157]: This: (($Boundaries) -or ($BoundariesToExclude))
Can be: ($Boundaries -or $BoundariesToExclude)

File: source/DSCResources/DSC_CMBoundaryGroups/DSC_CMBoundaryGroups.psm1 [hunk 3]
```diff
@@ -0,0 +1,364 @@
...
+
+                    foreach ($compare in $compares)
+                    {
+                        if ($Boundaries)
+                        {
+                            if ($compare.SideIndicator -eq '<=')
+                            {
+                                Write-Verbose -Message ($script:localizedData.ExcludingBoundary `
+                                    -f $BoundaryGroup, $compare.InputObject)
+                                Remove-CMBoundaryFromGroup -BoundaryName $compare.InputObject -BoundaryGroupName $BoundaryGroup
```
Comment (rchristman89) [inline source/DSCResources/DSC_CMBoundaryGroups/DSC_CMBoundaryGroups.psm1:180]: Check what happens if there are two with the same name.

File: source/DSCResources/DSC_CMBoundaryGroups/DSC_CMBoundaryGroups.psm1 [hunk 4]
```diff
@@ -0,0 +1,364 @@
...
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+    else
+    {
+        if ($state.Ensure -eq 'Present')
```
Comment (rchristman89) [inline source/DSCResources/DSC_CMBoundaryGroups/DSC_CMBoundaryGroups.psm1:352]: You can make the else an elseif:
elseif (($Ensure -eq 'Present') -and ($state.Ensure -eq 'Present'))

**Comment by jeffotterpohl** [comment]
Still need to update Pester tests for the helper function additions

**Comment by rchristman89** [comment]
When running Match on a boundary group that is missing the boundaries the verbose statement is not saying which group is missing.

**Pull request merged**